Model/ShimModel.php

```diff
@@ -931,4 +940,45 @@ public function alias($alias = null) {
 		return $this->alias;
 	}
 
+	/**
+	 * Returns the default validator object. Subclasses can override this function
+	 * to add a default validation set to the validator object.
+	 * This method allows to use Cake 3 syntax for defining validation rules
+	 * in models.
+	 * {@see https://book.cakephp.org/3.0/en/appendices/orm-migration.html#validation-no-longer-defined-as-a-property}
+	 *
+	 * @param Validator $validator The validator that can be modified to
+	 * add some rules to it.
+	 * @return Validator
+	 * @link https://api.cakephp.org/3.5/class-Cake.Validation.ValidatorAwareTrait.html#_validationDefault
+	 */
+	public function validationDefault(Validator $validator) {
+		Debugger::log($validator);
+		return $validator;
+	}
+
+	/**
+	 * Returns an instance of a model validator for this class.
+	 *
+	 * You can implement the method in `validationDefault` in your Model subclass
+	 * should you wish to have the default validation set.
+	 *
+	 * @param ModelValidator $instance Model validator instance.
```

Comment:
|null

Reply:
Are you sure? There is no `|null` [here](https://api.cakephp.org/2.10/source-class-Model.html#3895-3910). This is `@param`, not `@return`.

Reply:
Well, your code clearly says `(ModelValidator $instance = null)`

Reply:
So null is also allowed.